Return an empty array when no campaigns exist for GET ads/campaigns API

ianlin opened the pull request:
### Changes proposed in this Pull Request:

This PR fixes a bug that when no campaigns exist the `GET ads/campaigns` API will throw an error. It should return an empty array instead.

Closes #1364.

### Screenshots:

<img width="1250" alt="Screenshot 2022-03-25 at 17 57 23" src="https://user-images.githubusercontent.com/914406/160098617-be657351-75ea-4d0a-a16d-5f1b783e1e2e.png">

### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Remove all campaigns in your account.
2. Request to the API `GET ads/campaigns`.
3. The API should return an empty array.

### Additional details:

I was struggling to add a unit test for this case as this is an error from the Google API. Might be more suitable to have the integration test for it.

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

> Fix - Returns an empty array when no campaigns exist.

ianlin committed:
Return an empty array when no campaigns exist

ianlin committed:
Add a test for the case of returning empty array when no campaigns

mikkamp commented on tests/Unit/API/Google/AdsCampaignTest.php:
```diff
@@ -109,6 +114,10 @@ public function test_get_campaigns() {
 
 		$this->generate_ads_campaign_query_mock( $campaigns_data, $campaign_criterion_data );
+
+		$this->service_client->expects( $this->exactly( 2 ) )
+			->method( 'search' );
```
`service_client` is kind of part of the HelperTrait, so I think it would be better to move this expectation within the `generate_ads_campaign_query_mock` since that's the function that returns two results. Otherwise it looks a bit confusing what is expected in this test, and why it's not expected in the tests `test_get_campaigns_with_nonexist_location_id` and `test_get_campaigns_with_invalid_location_id`.

mikkamp replied:
Also maybe we can update the doc block for `generate_ads_campaign_query_mock` since it now generates mocked responses for two queries.

ianlin replied:
Thanks, good suggestions! Updated in 40fafee.

mikkamp commented on tests/Unit/API/Google/AdsCampaignTest.php:
```diff
@@ -70,6 +70,11 @@ public function setUp() {
 	public function test_get_campaigns_empty_list() {
 		$this->generate_ads_campaign_query_mock( [], [] );
+
+		// Method search() only being called by AdsCampaignQuery
+		$this->service_client->expects( $this->once() )
+			->method( 'search' );
```
Since `service_client` is part of HelperTrait I think it would be cleaner to create a separate helper function `generate_ads_campaign_query_mock_with_no_campaigns`. A comment can be added there that it only expects one of the queries to run.

ianlin replied:
Thanks, also updated in 40fafee.

mikkamp reviewed (approved):
Thanks for the changes, it's working as expected now.
Just left a small suggestion for making the tests a little cleaner.

ianlin committed:
Refactor tests for no campaigns case to make it cleaner

- Move expectation of `service_client->search` into the HelperTrait.
- Add a new helper method for the case of no campaigns.
- Update doc block.
